Examples/Swift/ViewController.swift

```diff
@@ -239,39 +218,38 @@ class ViewController: UIViewController, MGLMapViewDelegate {
     }
     
     // MARK: - Navigation with multiple waypoints
-
+    
     func startMultipleWaypoints() {
         guard let route = self.currentRoute else { return }
-
-        exampleMode = .multipleWaypoints
         
-        // When the user arrives at their destination, we'll prompt them to return back to where they started
-        nextWaypoint = self.currentRoute?.coordinates?.first
+        exampleMode = .multipleWaypoints
         
         let navigationViewController = NavigationViewController(for: route, locationManager: locationManager())
         navigationViewController.navigationDelegate = self
-
+        
         present(navigationViewController, animated: true, completion: nil)
     }
     
 }
 
 extension ViewController: NavigationViewControllerDelegate {
-    func navigationViewController(_ navigationViewController: NavigationViewController, didArriveAt destination: MGLAnnotation) {
+    func navigationViewController(_ navigationViewController: NavigationViewController, didArriveAt waypoint: Waypoint) {
         
         // Multiple waypoint demo
-        guard exampleMode == .multipleWaypoints, nextWaypoint != nil else { return }
-
+        guard exampleMode == .multipleWaypoints else { return }
+        
         // When the user arrives, present a view controller that prompts the user to continue to their next destination
         // This typ of screen could show information about a destination, pickup/dropoff confirmation, instructions upon arrival, etc.
         guard let confirmationController = self.storyboard?.instantiateViewController(withIdentifier: "waypointConfirmation") as? WaypointConfirmationViewController else { return }
-            
+        
         confirmationController.delegate = self
         
         navigationViewController.present(confirmationController, animated: true, completion: nil)
     }
-    
-    func navigationViewController(_ navigationViewController: NavigationViewController, didArriveAt waypoint: Waypoint) {
-        print("User arrived at \(waypoint.name ?? "waypoint")")
+}
+
+extension ViewController: WaypointConfirmationViewControllerDelegate {
+    func confirmationControllerDidConfirm(controller confirmationController: WaypointConfirmationViewController) {
```

Comment:
The first part of this selector, `confirmationControllerDidConfirm`, already contains “confirmation controller”, so `controller` can be `_`.

This code path would be a lot cleaner with an unwind segue instead of the delegate pattern…